src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -1214,6 +1217,45 @@ TEST_F(ClusterTaskManagerTest, HeartbeatTest) {
   }
 }
 
+TEST_F(ClusterTaskManagerTest, ResourceReportForNodeAffinitySchedulingStrategyTasks) {
```

Comment:
For clarity, it would be nice to include all 4 possibilities (worker node ID feasible, soft), so that we can see what the behavior should be in all cases.

Reply:
> Or maybe the code is already doing that and it's just not captured in test cases?

The code is already doing that, let me update the test cases.